src/main/java/application/FormController.java

```diff
@@ -341,7 +349,9 @@ public FormEntryResponseBean getNext(@RequestBody SessionRequestBean requestBean
                     formSession.getFormEntryModel(),
                     formSession.getCurrentIndex());
             updateSession(formSession, serializableFormSession);
-            FormEntryResponseBean responseBean = mapper.readValue(resp.toString(), FormEntryResponseBean.class);
+            FormEntryNavigationResponseBean responseBean = mapper.readValue(resp.toString(), FormEntryNavigationResponseBean.class);
+            responseBean.setIsAtLastIndex(formSession.getIsAtLastIndex());
```

Comment:
I'd move this logic into the getCurrentJson method which should allow you to remove some of the logic out of FormSession as well